lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -303,6 +308,30 @@ private void UpdateState(string taskId, TaskStateEvent taskEvent)
         }
 
         /// <summary>
+        /// This method is called when a running task is completed.
+        /// If it is master task, set _masterTaskCompleted to true
+        /// then calls RecordCompletedTask to remove the task from runningTasks and change task state.
+        /// </summary>
+        /// <param name="completedTask"></param>
+        internal void RecordCompletedRunningTask(ICompletedTask completedTask)
```

Comment:
I don't like that we have two methods to record completed task now. It's really confusing to figure out which one to call when. Can we add logic for handling `_masterTaskCompleted` to the old method?